src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class BaseFieldData:
+    """Contain common properties required by all field data types."""
+
+    def __init__(self, i_d, data):
+        self._data = data
+        self._id = i_d
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def surface_id(self):
+        return self._id
+
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(BaseFieldData):
+    """Container for scalar field data."""
+
+    class ScalarData:
+        """Store and provide the data as a scalar."""
+
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData.ScalarData(_data) for _data in data])
+
+
+class Vector:
+    """Stores the data in form of a vector (x, y, z)"""
+
+    def __init__(self, x, y, z):
+        self._x = x
+        self._y = y
+        self._z = z
+
+    @property
+    def x(self) -> float:
+        return self._x
+
+    @property
+    def y(self) -> float:
+        return self._y
+
+    @property
+    def z(self) -> float:
+        return self._z
+
+
+def _resolve_into_array_of_vectors(data):
+    if data.size % 3:
+        raise ValueError(
+            "Data set has to be resolved as set of vectors."
+            "The length of data set should always be in multiples of 3."
+        )
+    data.shape = data.size // 3, 3
+
+
+class VectorFieldData(BaseFieldData):
+    """Container for vector field data."""
+
+    class VectorData(Vector):
+        """Store and provide the data as vector."""
+
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data, scale):
+        _resolve_into_array_of_vectors(data)
+        self._scale = scale
+        super().__init__(i_d, [VectorFieldData.VectorData(x, y, z) for x, y, z in data])
+
+    @property
+    def scale(self) -> float:
+        return self._scale
+
+
+class Vertices(BaseFieldData):
+    """Container for Vertex data."""
+
+    class Vertex(Vector):
+        """Store and provide the data as a vector of Vertex."""
+
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        _resolve_into_array_of_vectors(data)
+        super().__init__(i_d, [(Vertices.Vertex(x, y, z)) for x, y, z in data])
+
+
+class FacesCentroid(BaseFieldData):
+    """Container for Faces Centroid data."""
```

Comment:
```suggestion
    """Provides the container for faces centroid data."""
```
I'm not sure here and in subsequent descriptions if faces should be possessive or if its plural. It seems like it should be face's centroid data., indicating the possessive form (where the centroid data belongs to a face.